### PR TITLE
Add OPFData dataset

### PR DESCRIPTION
**Pull request by mzgubic**
Hello PyG team!

We are releasing (it's public but not advertised yet) a large dataset for optimal power flow problems and want to make it easily loadable for torch users.

Cheers,
MIha

### CODE REVIEW

**Comment by mzgubic** [comment]
Hi @wsad1 , I've fixed the CI now, would you be able to review please?

File: torch_geometric/datasets/opfdata.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+def extract_edge_index_rev(json_file: dict, edge_name: str) -> torch.Tensor:
+    return torch.tensor([
+        json_file['grid']['edges'][edge_name]['receivers'],
+        json_file['grid']['edges'][edge_name]['senders'],
+    ])
+
+
+class OPFData(Dataset):
+    """Heterogeneous OPFData dataset from arxiv...TDOO...pdf.
```
Comment (mzgubic) [inline torch_geometric/datasets/opfdata.py:35]: This needs to be edited before merging.

**Comment by mzgubic** [comment]
Hi @rusty1s, thanks for taking a look.

How should I interpret being assigned this PR - is there something I should do before it can be reviewed?

**Comment by rusty1s** [comment]
All good, will merge ASAP :)

**Comment by mzgubic** [comment]
Thanks! I've added the arxiv link now, should be ready to go from my side too.

File: torch_geometric/datasets/opfdata.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+from typing import Callable, List, Literal, Optional, Tuple
+
+import torch
+
+from torch_geometric.data import Dataset, HeteroData, download_url, extract_tar
+
+
+def read_json(filename: str) -> dict:
+    """Reads json file."""
+    # with gfile.GFile(filename, 'r') as f:
```
Comment (mzgubic) [inline torch_geometric/datasets/opfdata.py:14]:
```suggestion
```

**Review by rusty1s — APPROVED** [review]

**Pull request merged**